### ISSUE TITLE
Suggestions for 1.0 milestone

### ISSUE THREAD

**Issue by adelevie** [adelevie/parse-ruby-client#76]
Discuss. Feel free to write the problem inline as a comment or just to link to an issue or PR.

Ideally, full parity with all of Parse's REST features would be nice. Wondering how others feel?

In some ways, I'm really happy with the stability of the codebase. Everything is pretty modular and magic-free. That said, some issues like delta-saving are still bugging me.

**Comment by rhymes** [comment]
My wishlist for the 1.0 is:

- remove Parse.init and @@client class variables
- add #169 and #170 
- fix as much bugs as possible

**Comment by Xavdidtheshadow** [comment]
Also, documentation (probably with YARD?) somewhere that isn't just the readme. README would be for instructions for development, some quick examples, and a link to common issues in the wiki (which is another thing to start developing).

**Comment by rhymes** [comment]
Another thing for 1.0:

 - wiki page explaining how to migrate from 0.3.0 to 1.0
 - wiki page explaining what's new

**Comment by Xavdidtheshadow** [comment]
@rhymes So I was thinking- I think the client change was the only big breaking change we're going to do- a lot of the other changes are features and stuff. 

Why don't we go ahead and cut the 1.0.0 release and then have all of the extra, currently missing parse features be minors after that? 

Alternatively, we could burn simple tasks like adding features TODO in the readme (such as deleting objets, etc) for a real v1 feeling, then cut the release?

**Comment by rhymes** [comment]
@Xavdidtheshadow Yep, we could do that, I feel that before though we should increase the test coverage a bit, review the doc (see also #190), write a changelog (we can use git-extras for that https://github.com/tj/git-extras/blob/master/Commands.md#git-changelog) and some sort of migration guide?

**Comment by Xavdidtheshadow** [comment]
Is there an easy place to see where we're missing test coverage on existing code?

Totally. I'll go through tomorrow and make a set of tasks we should actually get ready for 1.0 and when those are done, we can release!

**Comment by rhymes** [comment]
@Xavdidtheshadow only locally, I'll enable coveralls later on the repository so we can all see it. Thanks!

**Comment by rhymes** [comment]
@Xavdidtheshadow https://coveralls.io/github/adelevie/parse-ruby-client?branch=master here you are :D I also added the badge in the README

**Comment by Xavdidtheshadow** [comment]
@rhymes nice! that's way more than I thought it would be- it seems like we're in pretty good shape.

**Comment by rhymes** [comment]
yes indeed :)
